mackerel-plugin-php-fpm/lib/php-fpm_test.go

```diff
@@ -48,3 +48,96 @@ func TestGetStatus(t *testing.T) {
 	assert.EqualValues(t, 3, status.MaxListenQueue)
 	assert.EqualValues(t, 1000, status.SlowRequests)
 }
+
+func TestSocketFlag_Set(t *testing.T) {
+	tests := []struct {
+		Name string
+		URL  string
+
+		Network string
+		Address string
+		Err     string
+	}{
+		{
+			Name:    "parse absolute filepath",
+			URL:     "/dev/null",
+			Network: "unix",
+			Address: "/dev/null",
+		},
+		{
+			Name:    "parse relative filepath",
+			URL:     "a/b/c",
+			Network: "unix",
+			Address: "a/b/c",
+		},
+		{
+			Name:    "parse filename",
+			URL:     "file",
+			Network: "unix",
+			Address: "file",
+		},
+		{
+			Name:    "parse Go's address style",
+			URL:     "localhost:9000",
+			Network: "tcp",
+			Address: "localhost:9000",
+		},
+		{
+			Name:    "parse ipv6 address",
+			URL:     "[::1]:1000",
+			Network: "tcp",
+			Address: "[::1]:1000",
+		},
+		{
+			Name:    "parse unix:// scheme",
+			URL:     "unix:///path/to",
+			Network: "unix",
+			Address: "/path/to",
+		},
+		{
+			Name:    "parse tcp:// scheme",
+			URL:     "tcp://localhost",
+			Network: "tcp",
+			Address: "localhost", // should we set default port?
```

Comment:
TCP with no port should be a configuration error? I'm not sure...

Reply:
I deleted this comment.
PHP-FPM is listen on TCP/9000 by default.

> --with-fpm-port[=PORT]       Set the tcp port number to listen for cgi requests [9000]

Reply:
Nice.